package.json

```diff
@@ -240,7 +240,9 @@
     "@babel/core": "7.18.0",
     "@babel/parser": "7.18.0",
     "@babel/types": "7.18.0",
-    "@babel/traverse": "7.18.0"
+    "@babel/traverse": "7.18.0",
+    "@types/react": "18.0.37",
```

Comment:
Needed since `@mdx-js/react` still depends on 16.x types. Upgrading `@mdx-js/react` to 2.3 would pull in 18.x types but I wanted to focus this PR on types-only changes.

Without resolutions:
```bash
$ pnpm why @types/react
Legend: production dependency, optional only, dev only

nextjs-project@0.0.0 /home/eps1lon/Development/forks/next.js

devDependencies:
@mdx-js/react 2.2.1
└── @types/react 16.9.17
@next/mdx link:packages/next-mdx
└─┬ @mdx-js/react 2.2.1
  └── @types/react 16.9.17
@testing-library/react 13.0.0
└─┬ @types/react-dom 18.0.11
  └── @types/react 18.0.37
@types/react 18.0.37
@types/react-dom 18.0.11
└── @types/react 18.0.37
```